File: backend/app/schemas.py
from pydantic import BaseModel
from typing import Optional, Dict, Any
from datetime import datetime

class MenuSectionBase(BaseModel):
    name: str
    title: str
    description: Optional[str] = None
    icon: Optional[str] = None
    layout: str = "grid"
    sort_order: int = 0
    is_active: bool = True
    metadata: Dict[str, Any] = {}

class MenuSection(MenuSectionBase):
    id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class MenuItemBase(BaseModel):
    title: str
    description: Optional[str] = None
    icon: Optional[str] = None
    action_type: str
    action_value: Optional[str] = None
    parent_id: Optional[str] = None
    section_id: Optional[str] = None
    sort_order: int = 0
    is_active: bool = True
    is_premium: bool = False
    requires_auth: bool = False
    metadata: Dict[str, Any] = {}

class MenuItem(MenuItemBase):
    id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: admin_dashboard/src/lib/schemas.ts
import { z } from "zod"

export const menuItemSchema = z.object({
  id: z.string(),
  title: z.string().min(1),
  description: z.string().optional(),
  icon: z.string().optional(),
  action_type: z.enum(["screen", "url", "action", "section"]),
  action_value: z.string().min(1),
  parent_id: z.string().optional(),
  section_id: z.string().min(1),
  sort_order: z.number().int().min(0),
  is_active: z.boolean(),
  is_premium: z.boolean(),
  requires_auth: z.boolean(),
  meta_data: z.record(z.string(), z.any()).optional(),
})

export const menuSectionSchema = z.object({
  id: z.string(),
  name: z.string().min(1),
  title: z.string().min(1),
  description: z.string().optional(),
  icon: z.string().optional(),
  layout: z.enum(["grid", "list", "horizontal"]),
  sort_order: z.number().int().min(0),
  is_active: z.boolean(),
  meta_data: z.record(z.string(), z.any()).optional(),
})

export type MenuItem = z.infer<typeof menuItemSchema>
export type MenuSection = z.infer<typeof menuSectionSchema>